code/modules/pai/door_jack.dm

```diff
@@ -0,0 +1,134 @@
+#define CABLE_LENGTH 2
+
+/**
+ * Switch that handles door jack operations.
+ *
+ * @param {string} mode - The requested operation of the door jack.
+ *
+ * @returns {boolean} - TRUE if the door jack state was switched, FALSE otherwise.
+ */
+/mob/living/silicon/pai/proc/door_jack(mode)
+	if(isnull(mode))
+		return FALSE
+	switch(mode)
+		if(PAI_DOOR_JACK_CABLE)
+			extend_cable()
+			return TRUE
+		if(PAI_DOOR_JACK_HACK)
+			hack_door()
+			return TRUE
+		if(PAI_DOOR_JACK_CANCEL)
+			QDEL_NULL(hacking_cable)
+			visible_message(span_notice("The cable retracts into the pAI."))
+			return TRUE
+	return FALSE
+
+/**
+ * #Extend cable supporting proc
+ *
+ * When doorjack is installed, allows the pAI to drop
+ * a cable which is placed either on the floor or in
+ * someone's hands based (on distance).
+ *
+ * @returns {boolean} - TRUE if the cable was dropped, FALSE otherwise.
+ */
+/mob/living/silicon/pai/proc/extend_cable()
+	QDEL_NULL(hacking_cable) //clear any old cables
+	hacking_cable = new
+	var/mob/living/carbon/hacker = get_holder()
+	if(hacker?.put_in_hands(hacking_cable))
```

Comment:
why do you null check hacker here but not on the next few lines

Reply:
because the rest of the if block would not execute if hacker were null
basically this is "if there is a hacker and i can put something in their hands"